packages/vue-urql/src/useQuery.test.ts

```diff
@@ -0,0 +1,156 @@
+jest.mock('vue', () => {
+  const vue = jest.requireActual('vue');
+
+  return {
+    __esModule: true,
+    ...vue,
+    inject: () => client,
+  };
+});
+import { makeSubject } from 'wonka';
+import { createClient } from '@urql/core';
+import { useQuery } from './useQuery';
+import { nextTick, reactive } from 'vue';
+
+const client = createClient({ url: '/graphql', exchanges: [] });
+
+describe('useQuery', () => {
+  it('runs a query and updates data', async () => {
+    const subject = makeSubject<any>();
+    const executeQuery = jest
+      .spyOn(client, 'executeQuery')
+      .mockImplementation(() => subject.source);
+
+    const _query = useQuery({
+      query: `{ test }`,
+    });
+    const query = reactive(_query);
+
+    expect(query).toMatchObject({
+      data: undefined,
+      stale: false,
+      fetching: true,
+      error: undefined,
+      extensions: undefined,
+      operation: undefined,
+      isPaused: false,
+      pause: expect.any(Function),
+      resume: expect.any(Function),
+      executeQuery: expect.any(Function),
+      then: expect.any(Function),
+    });
+
+    expect(executeQuery).toHaveBeenCalledWith(
+      {
+        key: expect.any(Number),
+        query: expect.any(Object),
+        variables: {},
+        context: undefined,
+      },
+      {
+        pollInterval: undefined,
+        requestPolicy: undefined,
+      }
+    );
+
+    expect(query.fetching).toBe(true);
+
+    subject.next({ data: { test: true } });
+
+    expect(query.fetching).toBe(false);
+    expect(query.data).toEqual({ test: true });
+  });
+
+  /*
+   * This test currently fails for the following reasons:
+   * with pause: false, `executeQuery` is called 3 times
+   * with pause: true, the promise from useQuery never resolves
+   *
+   * it's unclear to me what the desired behaviour is supposed to be.
+   * - If we want to have the query execute only once, we would have to set `pause: true`,
+   *   but that seems counter-intuitive
+   * - If we don't pause, I'd expect 2 calls:
+   *   -  one from watchEffect
+   *   -  one from useQuery().then()
+   *   I can't say why it's 3 in the latter case-
+   *
+   */
+  it.skip('runs Query and awaits results', async () => {
+    const subject = makeSubject<any>();
+    const executeQuery = jest
+      .spyOn(client, 'executeQuery')
+      .mockImplementation(() => {
+        subject.next({ data: { test: true } });
+        return subject.source;
+      });
+
+    const query = await useQuery({
+      query: `{ test }`,
+      // pause: true,
```

Comment:
Oh that's interesting! I think this is because we can liken this behaviour to suspense in React, so once `useQuery`'s promise resolves and we subscribe again we don't have a result, since no `cacheExchange` is used.

Adding the appropriate exchanges however is not what we want though. While it would fix this, the cache doesn't cache errored results for instance, so instead we'll have to copy some code from `react-urql`'s `useQuery` where the result is cached just for the suspended stream for the next rerender after suspending.

I can take a look at fixing this

Reply:
My main problem here was that I wasn't sure what the use case of making it " thenable" is, so I wasn't sure what an appropriate fix would be.

Am I understanding correctly that this is supposed to support Vue's version of `Supsense`?

Reply:
Yea, I think I also don't quite know enough unfortunately quite yet to fix this. So it is indeed to support Vue's version of suspense, so we'd like to await the first result that `executeQuery` may yield but during normal execution we'd also like future results to still update the component.

I'm not quite clear on how the component lifecycle functions once Vue suspends, so it'd be interesting to know whether the lifecycle aborts when it does.

I think we may get into some trouble here with our `isThenable` flag, but essentially the idea was always to return the promise, resolve it on a result, and otherwise still keep a subscription active to update the result refs passively.

Reply:
Gotcha. I'll think a bit about it as well.